src/Inciter/AMR/Refinement_State.h

```diff
@@ -0,0 +1,167 @@
+//
+// Created by Robert Francis Bird on 10/27/16.
+//
+
+#ifndef TETAMR_AMR_DATA_H
+#define TETAMR_AMR_DATA_H
+
+#include <vector>
+#include <cstddef>
+
+#include "types.h"
+
+namespace AMR {
+    // Types
+    using real_t = tk::real;
```

Comment:
`tk::real` is defined in [Base/Types.h](https://github.com/quinoacomputing/quinoa/blob/master/src/Base/Types.h#L17). Thus I think this needs an `#include <Types.h>`, otherwise this header (as is right now if compiles) is order dependent. Or that is `types.h`? If so, let's combine!